tools/walletextension/rpcapi/blockchain_api.go

```diff
@@ -0,0 +1,271 @@
+package rpcapi
+
+//goland:noinspection ALL
+import (
+	"context"
+	"encoding/json"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ten-protocol/go-ten/go/common/gethapi"
+	"github.com/ten-protocol/go-ten/lib/gethfork/rpc"
+	wecommon "github.com/ten-protocol/go-ten/tools/walletextension/common"
+)
+
+type BlockChainAPI struct {
+	we *Services
+}
+
+func NewBlockChainAPI(we *Services) *BlockChainAPI {
+	return &BlockChainAPI{we}
+}
+
+func (api *BlockChainAPI) ChainId() *hexutil.Big { //nolint:stylecheck
+	// chainid, _ := UnauthenticatedTenRPCCall[hexutil.Big](nil, api.we, &CacheCfg{TTL: longCacheTTL}, "eth_chainId")
+	// return chainid
+	chainID := big.NewInt(int64(api.we.Config.TenChainID))
+	return (*hexutil.Big)(chainID)
+}
+
+func (api *BlockChainAPI) BlockNumber() hexutil.Uint64 {
+	nr, err := UnauthenticatedTenRPCCall[hexutil.Uint64](context.Background(), api.we, &CacheCfg{TTL: shortCacheTTL}, "eth_blockNumber")
+	if err != nil {
+		return hexutil.Uint64(0)
+	}
+	return *nr
+}
+
+func (api *BlockChainAPI) GetBalance(ctx context.Context, address common.Address, blockNrOrHash rpc.BlockNumberOrHash) (*hexutil.Big, error) {
+	// todo - how do you handle getBalance for contracts
+	return ExecAuthRPC[hexutil.Big](
+		ctx,
+		api.we,
+		&ExecCfg{
+			cacheCfg: &CacheCfg{
+				TTLCallback: func() time.Duration {
+					return cacheTTLBlockNumberOrHash(blockNrOrHash)
+				},
+			},
+			account:            &address,
+			tryUntilAuthorised: true, // the user can request the balance of a contract account
+		},
+		"eth_getBalance",
+		address,
+		blockNrOrHash,
+	)
+}
+
+// Result structs for GetProof
+type AccountResult struct {
+	Address      common.Address  `json:"address"`
+	AccountProof []string        `json:"accountProof"`
+	Balance      *hexutil.Big    `json:"balance"`
+	CodeHash     common.Hash     `json:"codeHash"`
+	Nonce        hexutil.Uint64  `json:"nonce"`
+	StorageHash  common.Hash     `json:"storageHash"`
+	StorageProof []StorageResult `json:"storageProof"`
+}
+
+type StorageResult struct {
+	Key   string       `json:"key"`
+	Value *hexutil.Big `json:"value"`
+	Proof []string     `json:"proof"`
+}
+
+/*
+	func (s *BlockChainAPI) GetProof(ctx context.Context, address common.Address, storageKeys []string, blockNrOrHash rpc.BlockNumberOrHash) (*AccountResult, error) {
```

Comment:
there are some apis which are present in go-ethereum, for which we don't have support yet.
I was thinking to leave them in, so we know. Maybe I should return "not implemented"

Reply:
Ok, good idea to leave them in so we have them in the comments in case we need to implement them.
I would just add a comment on top of that explaining that we don't support that yet / return non implemented. 
Otherwise it is a bit confusing to see some functions commented out without obvious reason.